core/message-handling.go

```diff
@@ -251,28 +255,56 @@ func makeMessageStreamHandler(handle incomingMessageHandler, logger *logging.Log
 	}
 }
 
+var forwardOut map[uint32](chan messages.Request)
+
 // startPeerConnections initiates asynchronous message exchange with
 // peer replicas.
 func startPeerConnections(replicaID, n uint32, connector api.ReplicaConnector, log messagelog.MessageLog, logger *logging.Logger) error {
 	supply := makePeerMessageSupplier(log)
+	forwardOut = make(map[uint32](chan messages.Request))
 
 	for peerID := uint32(0); peerID < n; peerID++ {
 		if peerID == replicaID {
 			continue
 		}
 
+		forwardOut[peerID] = make(chan messages.Request)
```

Comment:
Now we can also use `<-chan` here, Line 258, and 264.

Reply:
When I use `<-chan` on L258/L264/L271, compiler shows the following error on L302:

> invalid operation: cannot send to receive-only type <-chan messages.Request

https://github.com/Naoya-Horiguchi/minbft/blob/fe982365837a46d6326ac363937778ab10e70e8c/core/message-handling.go#L300-L303

L307 can have `<-` as you suggested, so I added it.

Reply:
> When I use <-chan on L258/L264/L271, compiler shows the following error on L302

Sorry, I overlooked the send operation in L302; They must be bidirectional.